File: programmers/Level 2/위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> spies;

    for(auto cloth: clothes)
        spies[cloth[1]] += 1;

    for(auto spy: spies){
        answer *= spy.second + 1;
    }
    return answer - 1;
}

File: programmers/Level 2/위장.py
from collections import defaultdict


def solution(clothes):
    answer = 1
    fashion_dict = defaultdict(int)
    for name, kind in clothes:
        fashion_dict[kind] += 1

    for key in fashion_dict.keys():  # 경우의 수
        answer *= (fashion_dict[key] + 1)

    return answer - 1
